docker/nginx/nginx.conf

```diff
@@ -0,0 +1,26 @@
+upstream graph-node {
+    server graph-node:8000;
+}
+
+map $http_origin $allow_origin {
+    default $http_origin;;
+}
+
+limit_req_zone $binary_remote_addr zone=applimit:10m rate=10r/s;
+limit_req_status 429;
+
+server {
+    listen 8000;
+    server_name $hostname;
+
+    client_max_body_size 16M;
+
+    location / {
+        proxy_pass http://graph-node/;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header Host $host;
+        proxy_redirect off;
+	    limit_req zone=applimit burst=20 nodelay;
```

Comment:
what is the difference between `limit_req` and `limit_req_zone`? are both needed?

Reply:
https://www.nginx.com/blog/rate-limiting-nginx/

Reply:
thanks sai. We discussed it on call and I explained what was in this blog to akshat as well.